api/v1alpha1/ndb_api_helpers.go

```diff
@@ -261,3 +251,77 @@ func GenerateDeprovisionDatabaseServerRequest() (req *DatabaseServerDeprovisionR
 	}
 	return
 }
+
+// Returns action arguments based on the type of database
+func GetActionArgumentsByDatabaseType(databaseType string) (DatabaseActionArgs, error) {
+	var dbTypeActionArgs DatabaseActionArgs
+	switch databaseType {
+	case DATABASE_TYPE_MYSQL:
+		dbTypeActionArgs = &MysqlActionArgs{}
+	case DATABASE_TYPE_POSTGRES:
+		dbTypeActionArgs = &PostgresActionArgs{}
+	case DATABASE_TYPE_MONGODB:
+		dbTypeActionArgs = &MongodbActionArgs{}
+	default:
+		return nil, errors.New("Invalid Database Type: supported values: mysql, postgres, mongodb")
+	}
+	return dbTypeActionArgs, nil
+}
+
+func (m *MysqlActionArgs) GetActionArguments(dbSpec DatabaseSpec) []ActionArgument {
+	return []ActionArgument{}
+}
+
+func (p *PostgresActionArgs) GetActionArguments(dbSpec DatabaseSpec) []ActionArgument {
+	return []ActionArgument{
+		{
+			Name:  "proxy_read_port",
```

Comment:
Consider defining these strings (eg listener_port) as constants in the constants file.